app/models/manageiq/providers/cisco_intersight/inventory/persister/definitions/physical_infra_collections.rb

```diff
@@ -2,21 +2,80 @@ module ManageIQ::Providers::CiscoIntersight::Inventory::Persister::Definitions::
   include ActiveSupport::Concern
 
   def initialize_physical_infra_collections
-    %i(
+    %i[
       physical_servers
       physical_server_details
       physical_server_computer_systems
       physical_racks
       physical_server_hardwares
       physical_server_network_devices
-      physical_server_firmwares      
+      physical_server_firmwares
       physical_server_storage_adapters
       physical_chassis
       physical_chassis_details
-    ).each do |name|
+      physical_switches
+      physical_switch_details
+      physical_switch_hardwares
+      physical_switch_firmwares
+    ].each do |name|
       add_collection(physical_infra, name)
 
+      add_physical_network_ports
+      add_physical_server_networks
+      add_physical_switch_networks
+      add_physical_server_management_devices
+    end
+  end
+
+  def add_physical_network_ports
+    %i[physical_server
+       physical_switch].each do |network_port_assoc|
+
+      add_collection(physical_infra, "#{network_port_assoc}_network_ports".to_sym) do |builder|
+        builder.add_properties(
+          :model_class                  => ::PhysicalNetworkPort,
+          :parent_inventory_collections => [network_port_assoc.to_s.pluralize.to_sym]
+        )
+
+        manager_ref = case network_port_assoc
+                      when :physical_server then %i[port_type uid_ems]
+                      when :physical_switch then %i[port_type port_name physical_switch]
+                      else []
+                      end
```

Comment:
Personal preference but this looks like overkill compared to just calling `add_collection` twice

Reply:
Collection physical_server_network_ports wasn't actually used in code, so I just removed its inclusion in the code above.